Description: Fix FTBFS with gcc 4.7 by fixing a bunch of missing includes.
Author: Cyril Brulebois <kibi@debian.org>
Bug-Debian: http://bugs.debian.org/667412

--- vowpal-wabbit-5.1+83-gffab10a.orig/v_array.h
+++ vowpal-wabbit-5.1+83-gffab10a/v_array.h
@@ -7,6 +7,8 @@ embodied in the content of this file are
 #ifndef VARRAY_H__
 #define VARRAY_H__
 #include <stdlib.h>
+#include <cstring>
+#include <algorithm>
 
 template<class T> class v_array{
  public:
@@ -43,7 +45,7 @@ template<class T> void push_many(v_array
   if(v.end+num >= v.end_array)
     {
       size_t length = v.end - v.begin;
-      size_t new_length = max(2 * (size_t)(v.end_array - v.begin) + 3, 
+      size_t new_length = std::max(2 * (size_t)(v.end_array - v.begin) + 3, 
 			      v.end - v.begin + num);
       v.begin = (T *)realloc(v.begin,sizeof(T) * new_length);
       v.end = v.begin + length;
--- vowpal-wabbit-5.1+83-gffab10a.orig/allreduce_master.cc
+++ vowpal-wabbit-5.1+83-gffab10a/allreduce_master.cc
@@ -17,6 +17,7 @@ This creates a binary tree topology over
 #include <string>
 #include <iostream>
 #include <cmath>
+#include <unistd.h>
 
 using namespace std;
 
--- vowpal-wabbit-5.1+83-gffab10a.orig/active_interactor.cc
+++ vowpal-wabbit-5.1+83-gffab10a/active_interactor.cc
@@ -6,6 +6,7 @@
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <netdb.h>
+#include <unistd.h>
 
 using std::cin;
 using std::endl;
--- vowpal-wabbit-5.1+83-gffab10a.orig/allreduce.cc
+++ vowpal-wabbit-5.1+83-gffab10a/allreduce.cc
@@ -21,6 +21,7 @@ Alekh Agarwal and John Langford, with he
 #include <string.h>
 #include <stdlib.h>
 #include <sys/timeb.h>
+#include <unistd.h>
 #include "allreduce.h"
 
 using namespace std;
--- vowpal-wabbit-5.1+83-gffab10a.orig/io.h
+++ vowpal-wabbit-5.1+83-gffab10a/io.h
@@ -8,6 +8,7 @@ embodied in the content of this file are
 #define IO_H__
 
 #include <fcntl.h>
+#include <unistd.h>
 #include "v_array.h"
 #include<iostream>
 
--- vowpal-wabbit-5.1+83-gffab10a.orig/network.cc
+++ vowpal-wabbit-5.1+83-gffab10a/network.cc
@@ -9,6 +9,7 @@
 
 #include <string>
 #include <iostream>
+#include <unistd.h>
 
 using namespace std;
 
